syntax = "proto3";

package secretmanagement;

option go_package = "internal/transport/grpc/proto";

message WriteSecretRequest {
  string secret = 1;
  string password = 2; 
}

message WriteSecretResponse {
  string first_token = 1;
  string second_token = 2;
}

message ReadSecretRequest {
  string first_token = 1;
  string second_token = 2;
}

message ReadSecretResponse {
  string secret = 1;
}

service SecretService {
  rpc WriteSecret(WriteSecretRequest) returns (WriteSecretResponse);
  rpc ReadSecret(ReadSecretRequest) returns (ReadSecretResponse);
}
